include/boost/url/detail/impl/recycler.ipp

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright (c) 2022 Vinnie Falco (vinnie.falco@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_DETAIL_IMPL_RECYCLER_IPP
+#define BOOST_URL_DETAIL_IMPL_RECYCLER_IPP
+
+#include <mutex>
+#include <utility>
+
+#ifdef _MSC_VER
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <debugapi.h>
+#endif
+
+namespace boost {
+namespace urls {
+namespace detail {
+
+struct all_reports
```

Comment:
Oh... I thought that was just for tests.

Reply:
These reports are for real programs. I wrote it this way to make sure that everything compiles, even if the reports are turned off (you have to define `BOOST_URL_REPORTS` to get results).